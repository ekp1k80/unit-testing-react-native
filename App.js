import React, {useState} from 'react';
import {
  SafeAreaView,
  StatusBar,
  StyleSheet,
  Text,
  TouchableOpacity,
  useColorScheme,
  View,
} from 'react-native';

import {Colors} from 'react-native/Libraries/NewAppScreen';

const App = () => {
  const [squareColor, setSquareColor] = useState(false);

  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
    flex: 1,
  };

  return (
    <SafeAreaView style={backgroundStyle}>
      <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
      <View style={styles.container}>
        <View
          style={squareColor ? styles.activeSquare : styles.inactiveSquare}
          testID="square"
        />
        <TouchableOpacity
          style={styles.button}
          onPress={() => setSquareColor(!squareColor)}
          activeOpacity={0.6}
          testID={'button'}>
          <Text style={styles.text}>Press Me</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  button: {
    width: '80%',
    height: '10%',
    backgroundColor: 'grey',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 100,
  },
  text: {
    fontSize: 20,
    color: 'white',
  },
  activeSquare: {
    width: 100,
    height: 100,
    backgroundColor: 'yellow',
  },
  inactiveSquare: {
    width: 100,
    height: 100,
    backgroundColor: 'black',
  },
});

export default App;
