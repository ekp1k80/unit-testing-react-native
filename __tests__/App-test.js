/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import TestRenderer, {act} from 'react-test-renderer';

describe('App', () => {
  const wrapper = TestRenderer.create(<App />);
  const instance = wrapper.root;

  it('renders correctly', () => {
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  describe('when button is not clicked', () => {
    let square = instance.findByProps({testID: 'square'});
    expect(square.props.style.backgroundColor).toBe('black');
  });

  describe('when button is clicked', () => {
    it('setSquareColor() is called with opposite value', () => {
      let square = instance.findByProps({testID: 'square'});
      const button = instance.findByProps({testID: 'button'});
      act(() => {
        button.props.onPress();
      });
      square = instance.findByProps({testID: 'square'});
      expect(square.props.style.backgroundColor).toBe('yellow');
    });
  });
});
