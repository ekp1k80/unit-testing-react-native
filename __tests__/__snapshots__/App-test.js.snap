// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App  renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#F3F3F3",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "black",
          "height": 100,
          "width": 100,
        }
      }
      testID="square"
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "grey",
          "display": "flex",
          "height": "10%",
          "justifyContent": "center",
          "marginTop": 100,
          "opacity": 1,
          "width": "80%",
        }
      }
      testID="button"
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 20,
          }
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
